src/main/java/org/altbeacon/beacon/service/scanner/CycledLeScanner.java

```diff
@@ -40,7 +42,10 @@
     private long mScanPeriod;
 
     protected long mBetweenScanPeriod;
-    protected final Handler mHandler = new Handler();
+
+    protected final Handler mHandler = new Handler(Looper.getMainLooper());
```

Comment:
The default, parameterless `Handler` constructor uses the `Looper` from the current thread (as documented [here](https://developer.android.com/reference/android/os/Handler.html#Handler()) and [here](https://developer.android.com/training/multiple-threads/communicate-ui.html)).

In this particular case, `CycledLeScanner` is always instantiated from the main thread (from `BeaconService.onCreate()`) and consequently, that `Handler` will already use the main thread's looper.
This addition doesn't modify the current logic, it's only a futureproof guarantee that the class behaviour won't change if in the future if `CycledLeScanner` is instantiated from another thread that doesn't have a `Looper` (you would immediately notice because the Handler creation would crash).